app/ui/src/main/java/de/mm20/launcher2/ui/launcher/widgets/weather/WeatherWidget.kt

```diff
@@ -430,6 +443,25 @@ fun WeatherDaySelector(
     }
 }
 
+private fun LazyListLayoutInfo.blendIntoViewAlpha(key: Any, degree: Float = 1f): Float =
+    visibleItemsInfo.firstOrNull { it.key == key }?.let {
+
+        val itemStart = it.offset
+        val itemEnd = it.offset + it.size
+
+        val atLeftEnd = itemStart < viewportStartOffset
+        val atRightEnd = itemEnd > viewportEndOffset
+
+        if (!atLeftEnd && !atRightEnd) {
+            return 1f
+        }
+
+        val alpha = 1f - (if (atLeftEnd) viewportStartOffset - itemStart else itemEnd - viewportEndOffset) / it.size.toFloat()
+
+        if (degree != 1f) alpha.pow(degree) else alpha
+
+    } ?: 1f
```

Comment:
Would this be something to put into core:ktx?

Reply:
I don't want Jetpack Compose dependencies in other modules so there is another `ktx` package in `de.mm20.launcher2.ui.ktx`, for extension functions for compose types.